packages/test/test-end-to-end-tests/src/test/flushModeValidation.spec.ts

```diff
@@ -0,0 +1,111 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { Container } from "@fluidframework/container-loader";
+import { SharedMap } from "@fluidframework/map";
+import { FlushMode } from "@fluidframework/runtime-definitions";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import {
+    ITestFluidObject,
+    ChannelFactoryRegistry,
+    ITestObjectProvider,
+    ITestContainerConfig,
+    DataObjectFactoryType,
+} from "@fluidframework/test-utils";
+import { describeNoCompat } from "@fluidframework/test-version-utils";
+
+/**
+ * This test validates that changing the FlushMode does not hit any validation errors in PendingStateManager.
+ * It also validates the scenario in this bug - https://github.com/microsoft/FluidFramework/issues/9398.
+ */
+describeNoCompat("Flush mode validation", (getTestObjectProvider) => {
+    const map1Id = "map1Key";
+    const registry: ChannelFactoryRegistry = [
+        [map1Id, SharedMap.getFactory()],
+    ];
+    const testContainerConfig: ITestContainerConfig = {
+        fluidDataObjectType: DataObjectFactoryType.Test,
+        registry,
+    };
+
+    let provider: ITestObjectProvider;
+    let dataObject1: ITestFluidObject;
+    let dataObject1map1: SharedMap;
+
+    async function ensureContainerConnected(container: Container): Promise<void> {
+        if (!container.connected) {
+            return new Promise((resolve) => container.once("connected", () => resolve()));
+        }
+    }
+
+    before(function() {
+        provider = getTestObjectProvider();
+        if (provider.driver.type !== "local") {
+            this.skip();
+        }
+    });
+
+    beforeEach(async () => {
+        // Create a Container for the first client.
+        const container1 = await provider.makeTestContainer(testContainerConfig) as Container;
+        dataObject1 = await requestFluidObject<ITestFluidObject>(container1, "default");
+        dataObject1map1 = await dataObject1.getSharedObject<SharedMap>(map1Id);
+        // Send an op in container1 so that it switches to "write" mode and wait for it to be connected.
+        dataObject1map1.set("key", "value");
+        await ensureContainerConnected(container1);
+        await provider.ensureSynchronized();
+    });
+
+    describe("Flush Mode validation", () => {
```

Comment:
Can we just move the tests in the scope above? This way, we can also ensure a reset of state between individual tests.

Reply:
done